OMCompiler/SimulationRuntime/c/simulation/solver/gbode_main.c

```diff
@@ -0,0 +1,1973 @@
+                                gbData->timeRight, gbData->yRight,
+                                gbfData->time + gbfData->lastStepSize, gbfData->y,
+                                gbData->nSlowStates, gbData->slowStates);
+    } else {
+      hermite_interpolation_gbf(gbData->timeLeft,  gbData->yLeft,  gbData->kLeft,
+                                gbData->timeRight, gbData->yRight, gbData->kRight,
+                                gbfData->time, gbfData->yOld,
+                                gbData->nSlowStates, gbData->slowStates);
+
+      hermite_interpolation_gbf(gbData->timeLeft,  gbData->yLeft,  gbData->kLeft,
+                                gbData->timeRight, gbData->yRight, gbData->kRight,
+                                gbfData->time + gbfData->lastStepSize, gbfData->y,
+                                gbData->nSlowStates, gbData->slowStates);
+    }
+
+    gbData->multi_rate_phase = 1;
+    eventTime = checkForEvents(data, threadData, solverInfo, gbfData->time, gbfData->yOld, gbfData->time + gbfData->lastStepSize, gbfData->y);
+    if (eventTime > 0)
+    {
+      solverInfo->currentTime = eventTime;
+      sData->timeValue = solverInfo->currentTime;
+
+      // sData->realVars are the "numerical" values on the right hand side of the event
+      gbData->time = eventTime;
+      memcpy(gbData->yOld, sData->realVars, gbData->nStates * sizeof(double));
+
+      gbfData->time = eventTime;
+      memcpy(gbfData->yOld, sData->realVars, gbData->nStates * sizeof(double));
+
+      /* write statistics to the solverInfo data structure */
+      solverInfo->solverStatsTmp[0] = gbfData->stats.nStepsTaken;
+      solverInfo->solverStatsTmp[1] = gbfData->stats.nCallsODE;
+      solverInfo->solverStatsTmp[2] = gbfData->stats.nCallsJacobian;
+      solverInfo->solverStatsTmp[3] = gbfData->stats.nErrorTestFailures;
+      solverInfo->solverStatsTmp[4] = gbfData->stats.nConvergenveTestFailures;
+
+      // log the emitted result
+      if (ACTIVE_STREAM(LOG_GBODE)){
+        infoStreamPrint(LOG_GBODE, 1, "Emit result (inner integration):");
+        printVector_gb(LOG_GBODE, " y", sData->realVars, nStates, sData->timeValue);
+        messageClose(LOG_GBODE);
+      }
+
+      if (ACTIVE_STREAM(LOG_GBODE_STATES)) {
+        dumpFastStates_gb(gbData, TRUE, eventTime);
+      }
+
+      // Get out of the integration routine for event handling
+      messageClose(LOG_SOLVER);
+      return 1;
+    }
+
+    /* update time with performed stepSize */
+    gbfData->time += gbfData->lastStepSize;
+
+    // store right hand values for latter interpolation
+    gbfData->timeRight = gbfData->time;
+    memcpy(gbfData->yRight, gbfData->y, nStates * sizeof(double));
+    // update kRight
+    sData->timeValue = gbfData->time;
+    memcpy(sData->realVars, gbfData->y, nStates * sizeof(double));
+    gbode_fODE(data, threadData, &(gbfData->stats.nCallsODE), fODE);
+    memcpy(gbfData->kRight, fODE, nStates * sizeof(double));
```

Comment:
`fODE` is not changing, it always points to `sData->realVars + data->modelData->nStates`